tests/javascript/frontediting/spec.js

```diff
@@ -0,0 +1,208 @@
+				jasmine.clock().install();
+				this.$editBtn.mouseleave();
+			});
+
+			afterEach(function () {
+				jasmine.clock().uninstall();
+			});
+
+			it('Should remove tooltip and button after a 500 millisecond delay', function () {
+				expect($('body>div.tooltip')).toExist();
+				expect($('body>a.btn.jmodedit')).toExist();
+
+				jasmine.clock().tick(501);
+
+				expect($('body>div.tooltip')).not.toExist();
+				expect($('body>a.btn.jmodedit')).not.toExist();
+			});
+		});
+	});
+
+	describe('Frontediting modules jmoddiv on mouseleave', function () {
+		beforeEach(function () {
+			jasmine.clock().install();
+			$('#jmoddiv-module').mouseenter();
+		});
+
+		afterEach(function () {
+			jasmine.clock().uninstall();
+		});
+
+		it('Should remove tooltip after a 500 millisecond delay', function () {
+			$('#jmoddiv-module').mouseleave();
+
+			expect($('body>a.btn.jmodedit')).toExist();
+
+			jasmine.clock().tick(501);
+
+			expect($('body>a.btn.jmodedit')).not.toExist();
+		});
+	});
+
+	describe('Frontediting menu jmoddiv li on mouseenter', function () {
+		beforeAll(function () {
+			$('#frontediting-menu-li').mouseenter();
+			this.$popover = $('body>div.popover');
+			this.$popoverContent = $('body>div.popover .btn.jfedit-menu');
+		});
+
+		it('Should show popover', function () {
+			expect(this.$popover).toExist();
+		});
+
+		it('Should have popover content an <a> tag having class btn jfedit-menu', function () {
+			expect(this.$popoverContent).toExist();
+		});
+
+		it('Should have href value of <a> element set', function () {
+			expect(this.$popoverContent).toHaveAttr('href', '/administrator/index.php?option=com_menus&view=item&layout=edit&id=123');
+		});
+
+		it('Should have target value of <a> element set to _blank', function () {
+			expect(this.$popoverContent).toHaveAttr('target', '_blank');
+		});
+
+		it('Should have title value of <a> element set to menu-title-123', function () {
+			expect(this.$popoverContent).toHaveAttr('data-original-title', 'menu-title-123');
+		});
+
+		describe('a.jfedit-menu on mouseenter', function () {
+			beforeAll(function () {
+				$('body>div.popover').find('a.jfedit-menu').mouseenter()
+			});
+
+			it('Should show tooltip', function () {
+				expect($('.popover-content div.tooltip')).toExist();
+			});
+		});
+
+		describe('popover on mouseleave', function () {
+			beforeAll(function () {
+				this.$popover.mouseleave();
+			});
+
+			it('Should show tooltip', function () {
+				expect($('body>div.popover')).not.toExist();
+			});
+		});
+	});
+
+	describe('Frontediting menu jmoddiv li on mouseleave', function () {
+		beforeEach(function () {
+			jasmine.clock().install();
+			$('#frontediting-menu-li').mouseenter();
+		});
+
+		afterEach(function () {
+			jasmine.clock().uninstall();
+		});
+
+		it('Should remove tooltip after a 500 millisecond delay', function () {
+			$('#frontediting-menu-li').mouseleave();
+
+			expect($('body>div.popover')).toExist();
+
+			jasmine.clock().tick(1501);
```

Comment:
Better to declare a variable/constant with the meaning for the tick e.g POPOVER_DELAY